src/main/scala/ch/epfl/bluebrain/nexus/kg/resolve/Resolver.scala

```diff
@@ -1,7 +1,95 @@
 package ch.epfl.bluebrain.nexus.kg.resolve
 
-trait Resolver {
+import java.time.Instant
 
+import ch.epfl.bluebrain.nexus.kg.config.Vocabulary.nxv
+import ch.epfl.bluebrain.nexus.kg.resources.{ProjectRef, ResourceV}
+import ch.epfl.bluebrain.nexus.rdf.Iri.AbsoluteIri
+import ch.epfl.bluebrain.nexus.rdf.syntax.node._
+
+import scala.util.Try
+
+/**
+  * Enumeration of Resolver types.
+  */
+sealed trait Resolver extends Product with Serializable {
+
+  /**
+    * @return a reference to the project that the resolver belongs to
+    */
+  def ref: ProjectRef
+
+  /**
+    * @return the resolver id
+    */
+  def id: AbsoluteIri
+
+  /**
+    * @return the resolver revision
+    */
+  def rev: Long
+
+  /**
+    * @return the instant when the resolver revision was created
+    */
+  def instant: Instant
+
+  /**
+    * @return the deprecation state of the resolver
+    */
+  def deprecated: Boolean
+
+  /**
+    * @return the resolver priority
+    */
   def priority: Int
+}
+
+object Resolver {
+
+  /**
+    * Attempts to transform the resource into a [[ch.epfl.bluebrain.nexus.kg.resolve.Resolver]].
+    *
+    * @param resource a materialized resource
+    * @return Some(resolver) if the resource is compatible with a Resolver, None otherwise
+    */
+  final def apply(resource: ResourceV): Option[Resolver] =
+    if (resource.types.contains(nxv.Resolver.value)) {
+      resource.value.graph.cursor(resource.id.value).downField(_ == nxv.priority).values match {
```

Comment:
The `id` is part of the graph, but since we already know the it, it's simpler to just use it instead of trying to detect it; it should also be faster.